crates/daphne-server/src/roles/leader.rs

```diff
@@ -165,16 +140,30 @@ impl crate::App {
                 .map_err(|e| fatal_error!(err = ?e, "failed to construct content-type header"))?,
         );
 
-        if let Some(bearer_token) = req.sender_auth.and_then(|auth| auth.bearer_token) {
+        if let Some(auth) = self
+            .service_config
+            .leader_to_helper_tokens
+            .get(&url.host().unwrap().to_owned())
```

Comment:
I don't think this unwrap is safe. From the documentation:
```
    /// Return the parsed representation of the host for this URL.
    /// Non-ASCII domain labels are punycode-encoded per IDNA if this is the host
    /// of a special URL, or percent encoded for non-special URLs.
    ///
    /// Cannot-be-a-base URLs (typical of `data:` and `mailto:`) and some `file:` URLs
    /// don’t have a host.
```

Reply:
We're unlikely to panic here, but it seems like a good idea to handle as a fatal error.

Reply:
I think it's okay to panic here, this url should originate from trusted input (configuration) and not from external input

Reply:
IMO we shouldn't treat configuration as valid input as a general matter. In this particular case, I think the `Url` parser would be happy to consume a URL that doesn't have a host.